rfd/0047-repository-management-features.md

```diff
@@ -0,0 +1,274 @@
+
+This portion of the bot runs as a CRON job on an interval of 30 minutes. 
+
+```yaml
+  #  Example dismissing stale runs workflow 
+  name: Dismiss Stale Workflows Runs
+on:
+  schedule:
+    - cron:  '0,30 * * * *'
+     
+jobs: 
+  dismiss-stale-runs:
+    name: Dismiss Stale Workflow Runs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the master branch 
+        uses: actions/checkout@master
+      - name: Installing the latest version of Go.
+        uses: actions/setup-go@v2
+        # Run "dismiss-runs" subcommand on bot.
+      - name: Dismiss
+        run: cd .github/workflows/teleport-ci && go run cmd/bot.go --token=${{ secrets.GITHUB_TOKEN }} dismiss-runs
+```
+
+--- 
+
+## Backporting Docs
+
+**Status:** _In Progress_ 
+
+###  What 
+
+Automatically backport changes to `docs/` to the specified branches. 
+
+### Why 
+
+Backporting docs is time consuming and there are often many doc changes. Implementing a way to automate backporting will increase the efficiency and speed of the development team. 
+
+### Implementation
+
+Command: `backport`
+
+
+#### Informing the Bot
+
+When there is a pull request opened and needs to be backported, the author will tag the `github-actions` bot with the string "Backport" and following will be a comma separated list of the branches to backport to.  Example backport comment: 
+
+``` 
+ @github-actions Backport branch-1, branch-2
+```
+
+#### Backport Requests
+
+The workflow will trigger anytime the `issue_comment` event occurs. 
+
+The `issue_comment` event occurs for comments on both issues and pull requests. To determine whether the `issue_comment` event was triggered from a pull request, the event payload will have an  `issue.pull_request` property and we can use it as a condition to only run the workflow on pull requests comment events.
+
+```yaml
+# docs-backport.yml
+on:
+  issue_comment:
+    types: [created]
+  jobs:
+    merge_job:
+      # This will only run for pull request comments
+      if: ${{ github.event.issue.pull_request }}
+      runs-on: ubuntu-latest
+      # jobs...
+            
+```
+The backport bot will only support merged pull requests. 
+
+
+#### Backporting 
+
+The process of backporting a branch is checking out the targeted branch and using `git cherry-pick` with the merge commit. The following process will happen for every branch label in the pull request. The bot will use the `os/exec` package to checkout branches and `cherry-pick` the commit.  
+
+_A potential library to use: [`git2go`](https://github.com/libgit2/git2go) which is essentially a Go wrapper for `libgit2`._
```

Comment:
I would use a library and not execute commands on disk unless last resort.

Reply:
I actually think at this point executing commands on disk is necessary. For the following reasons that I've found: 
- `libgit2` on the Github Actions runner is an older version and it seems I can't install a newer version to work with a version of `git2go` that can properly authenticate with Github (found a similar issue the .NET library is having with libgit2  https://github.com/NuKeeperDotNet/NuKeeper/issues/1042). I considered creating a custom image for this with a newer version of `libgit2` that *may* work, but that seems excessive considering this is just an issue with 1 library. 
- `git2go` is poorly documented and can't really seem to get anything going with it. @tcsc can attest to this.